etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,516 @@
+	// +optional
+	Spec *PacemakerClusterSpec `json:"spec,omitempty"`
+
+	// status contains the actual pacemaker cluster status information collected from the cluster.
+	// This follows the Design Principle: Act on Deterministic Information.
+	// When fields are not present, they are treated as unknown and no actions are taken by the cluster-etcd-operator.
+	// +optional
+	Status *PacemakerClusterStatus `json:"status,omitempty,omitzero"`
+}
+
+// PacemakerClusterSpec is an empty spec as PacemakerCluster is a status-only resource
+type PacemakerClusterSpec struct {
+}
+
+// PacemakerClusterStatus contains the actual pacemaker cluster status information
+// As part of validating the status object, we need to ensure that the lastUpdated timestamp is always newer than the current value.
+// We allow the lastUpdated timestamp to be empty on initial creation, but it is required once it has been set.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.lastUpdated) || (has(self.lastUpdated) && self.lastUpdated >= oldSelf.lastUpdated)",message="lastUpdated must be a newer timestamp than the current value"
+type PacemakerClusterStatus struct {
+	// lastUpdated is the timestamp when this status was last updated
+	// When present, it must be a valid timestamp in RFC3339 format and cannot be set to an earlier timestamp than the current value
+	// This is the most critical field in the status object because we can use it to warn if the status collection has gone stale.
+	// It is optional upon initial creation, but required once it has been set.
+	// +kubebuilder:validation:Format=RFC3339
+	// +optional
+	LastUpdated metav1.Time `json:"lastUpdated,omitempty"`
+
+	// rawXML contains the raw XML output from pcs status xml command.
+	// Kept for debugging purposes only; healthcheck should not need to parse this.
+	// When present, it must be between 1 and 262144 characters long (max 256KB). This limit protects the API from XML bombs and excessive memory consumption.
+	// When not present, the raw XML output is not available.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=262144
+	// +optional
+	RawXML string `json:"rawXML,omitempty"`
+
+	// collectionError contains any error encountered while collecting status
+	// When present, it must be between 1 and 2048 characters long (max 2KB). This limit ensures that the error message can be displayed in a human-readable format.
+	// When not present, no collection errors are available and the status collection is assumed to be successful.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	CollectionError string `json:"collectionError,omitempty"`
+
+	// summary provides high-level counts and flags for the cluster state
+	// TNF clusters don't use advanced pacemaker features like maintenance mode or pacemaker remote nodes, so these are omitted from the summary.
+	// When present, it must be a valid PacemakerSummary object.
```

Comment:
This doesn't add much value for an end user does it?

Reply:
Removed in favor of conditions the users may need to care about.